### DISCUSSION TITLE
Build failed

### DISCUSSION THREAD

**Discussion by 28arnab** [LazyVim/LazyVim#4194] [Q&A] [answered]
Failed (1)
    ○ markdown-preview.nvim  MarkdownPreview  MarkdownPreviewStop  MarkdownPreviewToggle  <leader>cp 
        Vim:E117: Unknown function: mkdp#util#install

![image](https://github.com/user-attachments/assets/6ff60c98-496d-4620-8895-d4308444d008)

**Comment by dpetka2001** [comment]
It seems that something is broken with the latest versions of NodeJS. Putting the following in your personal configuration seems to bypass this problem
```lua
  {
    "iamcco/markdown-preview.nvim",
    build = "cd app && npm install",
  },
```

**Reply by 28arnab** [reply]
where is this personal config file i just downloaded lazy idk this
![image](https://github.com/user-attachments/assets/37d0a410-ccd5-457a-b735-1b4c0f383a85)

**Reply by dpetka2001** [reply, marked as answer]
Create a file in `~/.config/nvim/lua/plugins/markdown.lua` with contents
```lua
return {
  {
    "iamcco/markdown-preview.nvim",
    build = "cd app && npm install",
  },
}
```

**Reply by dpetka2001** [reply]
I've created a PR #4196 to address this. You might want to change the `build` line into
```lua
build = function()
      require("lazy").load({ plugins = { "markdown-preview.nvim" } })
      vim.fn["mkdp#util#install"]()
end,
```
because `npm` will make changes in the local git repo of the plugin and you might encounter errors when trying to check for updates and stuff via `Lazy`.
